model.go

```diff
@@ -16,11 +16,12 @@ type ModelInfo struct {
 	Provider    string `json:"provider"`
 	Host        string `json:"host"`
 	Path        string `json:"path"`
+	Mode        string `json:"mode"`
 }
 
 func (m ModelInfo) Hash() string {
 	// Concatenate the attributes into a single string
-	data := fmt.Sprintf("%v|%v|%v|%v", m.Name, m.SourceModel, m.Provider, m.Host)
+	data := fmt.Sprintf("%v|%v|%v|%v|%v", m.Name, m.SourceModel, m.Provider, m.Host, m.Mode)
```

Comment:
Not sure that we want the mode as part of the hash.  It doesn't help to uniquely identify the model.

Reply:
i think we do, since we're using the hash to decide whether or not to update the `ModelInstance` on the DB right. so if we don't put it as part of the hash, when someone changes their model from one mode to the other, it would be a NOOP and the DB won't be updated.

https://github.com/hypermodeAI/monorepo/blob/main/api/githubapp/sync_db_manifest.go#L117

Reply:
Perhaps then only include it in the hash if `shared=true` (or `dedicated=false`)? I'm just thinking that when this rolls out, that existing non-shared models should have the same hash as before.

Reply:
ah good point, we should do that. similarly, we should also check if `host='hypermode'`, so non hypermode-hosted models are not affected.